infra/scripts/bash-helpers.sh

```diff
@@ -49,44 +79,52 @@ vic-create () {
-    if [ -z "${DOCKER_TLS_VERIFY+x}" ] && [ -z "${TLS_OPTS+x}" ]; then
+    if [ -z "${DOCKER_TLS_VERIFY+x}" ] && [[ "${DOCKER_HOST+x}" != "*:2375" ]]; then
         alias docker='docker --tls'
     fi
-
-    cd "$base" || exit
-}
-
-vic-delete () {
-    "$(vic-path)"/bin/vic-machine-"$OS" delete --target="$GOVC_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --thumbprint="$THUMBPRINT" --force "$@"
 }
 
 vic-inspect () {
-    "$(vic-path)"/bin/vic-machine-"$OS" inspect --target="$GOVC_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --thumbprint="$THUMBPRINT" "$@"
+    vicProfileTranscode
+
+    # change to the bin directory as that's where our certs would have been generated by vic-create
+    (
+        cd "$(vic-path)"/bin || return
+        "$(vic-path)"/bin/vic-machine-"$OS" inspect --target="$TARGET_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --thumbprint="$THUMBPRINT" "$@"
+    )
 }
 
 vic-upgrade () {
-    "$(vic-path)"/bin/vic-machine-"$OS" upgrade --target="$GOVC_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --thumbprint="$THUMBPRINT" "$@"
+    vicProfileTranscode
+
+    "$(vic-path)"/bin/vic-machine-"$OS" upgrade --target="$TARGET_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --thumbprint="$THUMBPRINT" "$@"
 }
 
 vic-ls () {
-    "$(vic-path)"/bin/vic-machine-"$OS" ls --target="$GOVC_URL" --thumbprint="$THUMBPRINT" "$@"
+    vicProfileTranscode
+
+    "$(vic-path)"/bin/vic-machine-"$OS" ls --target="$TARGET_URL" --thumbprint="$THUMBPRINT" "$@"
 }
 
 vic-ssh () {
+    vicProfileTranscode
+
     unset keyarg
     if [ -e "$HOME"/.ssh/authorized_keys ]; then
         keyarg="--authorized-key=$HOME/.ssh/authorized_keys"
     fi
 
-    out=$("$(vic-path)"/bin/vic-machine-"$OS" debug --target="$GOVC_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --enable-ssh "$keyarg" --rootpw=password --thumbprint="$THUMBPRINT" "$@")
+    out=$("$(vic-path)"/bin/vic-machine-"$OS" debug --target="$TARGET_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --enable-ssh $keyarg --rootpw=password --thumbprint="$THUMBPRINT" "$@")
     host=$(echo "$out" | grep DOCKER_HOST | awk -F"DOCKER_HOST=" '{print $2}' | cut -d ":" -f1 | cut -d "=" -f2)
 
     echo "SSH to ${host}"
     sshpass -ppassword ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@"${host}"
 }
 
 vic-admin () {
-    out=$("$(vic-path)"/bin/vic-machine-"$OS" debug --target="$GOVC_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --enable-ssh "$keyarg" --rootpw=password --thumbprint="$THUMBPRINT" "$@")
+    vicProfileTranscode
+
+    out=$("$(vic-path)"/bin/vic-machine-"$OS" debug --target="$TARGET_URL" --compute-resource="$COMPUTE" --name="${VIC_NAME:-${USER}test}" --enable-ssh "$keyarg" --rootpw=password --thumbprint="$THUMBPRINT" "$@")
```

Comment:
I see that you removed double quotation at some places and at some didn't. It is on purpose?

Reply:
I don't see any quotes changed - that said I _do_ see that `$keyarg` is incorrectly quoted meaning you cannot have an empty authorized_keys file.

Reply:
Ah, I see you mean the other functions. Yeah, none of them should have keyarg quoted.